syntax = "proto3";

service Env {
  rpc Make (Name) returns (Info) {};
  rpc Reset (Empty) returns (Observation) {};
  rpc Step (Action) returns (Transition) {};
}

message Name {
  string value = 1;
}

message Info {
  repeated int32 observation_shape = 1;
  int32 num_actions = 2;
  int32 max_episode_steps = 3;
}

message Action {
  int32 value = 1;
}

message Observation {
  repeated float data = 1;
  repeated int32 shape = 2;
}

message Transition {
  Observation observation = 1;
  float reward = 2;
  Observation next_episode = 3;
}

message Empty {}
